usvm-python/usvm-python-main/src/main/kotlin/org/usvm/machine/types/VirtualTypes.kt

```diff
@@ -113,42 +114,42 @@ object HasMpLength : TypeProtocol() {
         ConcretePythonInterpreter.typeHasMpLength(type.asObject)
 }
 
-object HasMpSubscript : TypeProtocol() {
+object HasMpSubscript : TypeProtocol(SlotId.MpSubscript) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasMpSubscript(type.asObject)
 }
 
-object HasMpAssSubscript : TypeProtocol() {
+object HasMpAssSubscript : TypeProtocol(SlotId.MpAssSubscript) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasMpAssSubscript(type.asObject)
 }
 
-object HasTpRichcmp : TypeProtocol() {
+object HasTpRichcmp : TypeProtocol(SlotId.TpRichcompare) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasTpRichcmp(type.asObject)
 }
 
-object HasTpGetattro : TypeProtocol() {
+object HasTpGetattro : TypeProtocol(SlotId.TpGetattro) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasTpGetattro(type.asObject)
 }
 
-object HasTpSetattro : TypeProtocol() {
+object HasTpSetattro : TypeProtocol(SlotId.TpSetattro) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasTpSetattro(type.asObject)
 }
 
-object HasTpIter : TypeProtocol() {
+object HasTpIter : TypeProtocol(SlotId.TpIter) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasTpIter(type.asObject)
 }
 
-object HasTpCall : TypeProtocol() {
+object HasTpCall : TypeProtocol(SlotId.TpCall) {
     override fun acceptsConcrete(type: ConcretePythonType): Boolean =
         ConcretePythonInterpreter.typeHasTpCall(type.asObject)
 }
 
-object HasTpHash : TypeProtocol() {
+object HasTpHash : TypeProtocol(SlotId.TpHash) {
```

Comment:
Please add this comment for mandatory slots:

```
TODO: since we cannot turn off this slot on virtual object, we may need to remove this [TypeProtocol] in the future.
```